### PR TITLE
Using native JSON operations to add thread participants

### PR DESCRIPTION
**Pull request by agnivade**
We use DB native JSON operations to add thread participants.

https://community-daily.mattermost.com/plugins/focalboard/workspace/zyoahc9uapdn3xdptac6jb69ic?id=285b80a3-257d-41f6-8cf4-ed80ca9d92e5&v=495cdb4d-c13a-4992-8eb9-80cfee2819a4&c=c174b031-1b1b-4425-8184-4bc6b1bf6059

```release-note
NONE
```

### CODE REVIEW

File: store/sqlstore/store.go [hunk 1]
```diff
@@ -163,6 +165,30 @@ type ColumnInfo struct {
 	CharMaximumLength int
 }
 
+type jsonArray []string
```
Comment (agnivade) [inline store/sqlstore/store.go:168]: Let me know if you feel this should be moved to a new file. I think this file has accumulated a lot of cruft that can be moved to an adapters.go file or sth.

Comment (streamer45) [reply store/sqlstore/store.go:168]: Yeah that would be a great idea. I'd also add a small unit test for `Value()`, just for correctness sake.

File: store/sqlstore/thread_store.go [hunk 2]
```diff
@@ -658,14 +659,21 @@ func (s *SqlThreadStore) MaintainMembership(userId, postId string, opts store.Th
...
-		}
-		if thread != nil && !thread.Participants.Contains(userId) {
-			thread.Participants = append(thread.Participants, userId)
-			if _, err = s.update(trx, thread); err != nil {
-				return nil, err
+		if s.DriverName() == model.DatabaseDriverPostgres {
+			if _, err2 := trx.Exec(`UPDATE Threads
+				SET participants = participants || $1::jsonb
+				WHERE postid=$2
+				AND not participants ? $3`, jsonArray([]string{userId}), postId, userId); err2 != nil {
```
Comment (streamer45) [inline store/sqlstore/thread_store.go:666]: As a personal favor, could we make `not` uppercase? :p

**Review by streamer45 — APPROVED** [review]
Lovely :+1: Just left a couple of non-blocking comments.

File: store/sqlstore/thread_store.go (hunk 2 above)
Comment (agnivade) [reply store/sqlstore/thread_store.go:666]: Sure, it was bothering me too :P

**Review by ashishbhate — APPROVED** [review]

**Pull request merged**